weave/workers/weave_scorer.py

```diff
@@ -0,0 +1,445 @@
+import asyncio
+import inspect
+import logging
+import os
+from datetime import datetime
+from typing import Any, Optional
+
+import sentry_sdk
+from confluent_kafka import KafkaError, Message
+from tenacity import (
+    before_log,
+    retry,
+    retry_if_exception_type,
+    stop_after_attempt,
+    wait_fixed,
+)
+
+# This import is used to register built-in scorers so they can be deserialized from the DB
+import weave.scorers  # noqa: F401
+from weave.flow.monitor import Monitor
+from weave.flow.scorer import Scorer, get_scorer_attributes
+from weave.trace.box import box
+from weave.trace.objectify import maybe_objectify
+from weave.trace.op import _default_on_input_handler
+from weave.trace.serialization.serialize import to_json
+from weave.trace.weave_client import (
+    RUNNABLE_FEEDBACK_TYPE_PREFIX,
+    Call,
+    FeedbackCreateReq,
+    ObjectRef,
+    from_json,
+)
```

Comment:
All these require the SDK code to be included in the scorer Docker image, which means a dedicated image needs to be built, instead of reusing the server image, which would be ideal (lighter, simpler build process, simpler Helm charts).

Therefore, either we are ok with the server image containing SDK code, or we could refactor the SDK to isolate components that are required both in the SDK and the worker (i.e. the above imports).
For example, we could have `weave.lib` module that contains primitives (e.g. `Call`, `Monitor`, `Scorer`, etc.) and could be harmlessly importer in both the SDK and the workers.
Then a single image would be sufficient for all backend processes (server, worker(s), etc.). Much cleaner.

Reply:
I actually ended up copying the entire SDK code into the server image so that I can resuse for the worker. The build process and helm charts are much simpler now. See https://github.com/wandb/core/pull/29039

If separation of client and backend code is very important, isolate `weave.lib` as state above.

Reply:
agreed - I think this is the way to go for now. We will probably want to split things up in the future